[En] add Host

jasonrhodes opened the pull request:
### Describe your changes

This has been a long time coming after I initially submitted this idea years ago!

As best as I can tell, this is similar to the "service" / "microservice" relationship, so I've tried to model this doc after the "service" one. I've added a bit of context about "host" that isn't present in "nodes", but maybe it's unnecessary? Feedback definitely appreciated.

### Related issue number or link (ex: `resolves #issue-number`)

Resolves #1308

### Checklist before opening this PR (put `x` in the checkboxes)
- [x] This PR does not contain plagiarism
  - don’t copy other people’s work unless you are quoting and contributing it to them.
- [x] I have signed off on all commits 
  - [signing off](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) (ex: `git commit -s`) is to affirm that commits comply [DCO](https://wiki.linuxfoundation.org/dco). If you are working locally, you could add an alias to your `gitconfig` by running `git config --global alias.ci "commit -s"`.

iamNoah1 reviewed (changes requested):
I really like the simplicity of this description. However, usually we have a certain structure that we follow for this category., ie: Problem it Addresses and How it helps. But for this term I don't really see how we should press into this template ^^, I am curious if some other maintainer has an idea how to proceed. 

Appart from that the content is nice, just have a look at the failing spell and link check. 

cc @hlipsig @jihoon-seo @seokho-son @Okabe-Junya

jasonrhodes commented:
I've added semantic line breaks and fixed the failing "LINKS" check by adding the missing trailing slash to the nodes link. The spellcheck is failing due to the use of "hostname", which is the word I intended to use. Is there a way to specify that it's valid or does it need to be rewritten to "host name" just to pass this check?

Thanks!

nate-double-u commented:
> I've added semantic line breaks and fixed the failing "LINKS" check by adding the missing trailing slash to the nodes link. 

Thanks!

> The spellcheck is failing due to the use of "hostname", which is the word I intended to use. Is there a way to specify that it's valid or does it need to be rewritten to "host name" just to pass this check?

You can add it to the `wordlist.txt` file.

jasonrhodes commented:
> You can add it to the wordlist.txt file.

@nate-double-u got it, I've added "hostname" there, thanks.

> However, I think it's also necessary to provide a consistent style for all terminology so that users don't get confused

@Okabe-Junya I definitely agree. That was why I modeled this change after the other "redirect" style of definition in the glossary already. Please take a look at [the current service definition file](https://github.com/cncf/glossary/blob/821376ed090914b47cd415c79f0e6b88d0a21b4b/content/en/service.md). If you all would like these "redirect" definition files to include the same sections as the others, I'd like some guidance on how to approach it. Copy/paste from the target? 

Thanks.

jasonrhodes commented:
The last line of this new "host" file and of the mentioned "service" file both use "Please see the [target] definition", these could be expanded to say "For more about the problem it addresses and how it helps, please see the [target] definition" if that helps tie the formats together? Interested in more feedback here.

Okabe-Junya commented:
Thank you for pointing out, Jason, Noah!

I see... The "Problem it addresses" section is indeed very difficult. I've thought about this term many times, and I think that "host" would be a conceptual/abstract term rather than a term for solving specific problems (as you all say, that's already explained by other existing terms).

How about an option where we keep only the "How it helps" section and omit the "Problem it addresses" section?

Thank you again for your contribution and discussion :)

jihoon-seo commented:
IMO, while it's important to follow the established template, there may be terms that are difficult to fill in the "Problem it addresses" section and the accompanying "How it helps" section in the template, and I think the term "Host" that Jason is trying to add in this PR is one of those. So I think it would be nice to allow exceptions when necessary.. :)

Okabe-Junya commented:
Sorry for the late reply.

It looks like Nate and Seokho reacted "+1" for Noah’s comment. 
Let’s move this PR forward by omitting that section :)

(I'll review this PR later)

iamNoah1 reviewed (approved):
@jasonrhodes I am happy to approve now that we have a consensus amongst maintainers. Thanks again for the effort :)
